video/out/vo.c

```diff
@@ -597,6 +597,7 @@ static void run_reconfig(void *p)
     mp_mutex_lock(&vo->params_mutex);
     talloc_free(vo->params);
     vo->params = talloc_dup(vo, params);
+    vo->target_params = NULL;
```

Comment:
what's the good reason for doing this?
this is the exact inverse of my reasoning in 2893b7d0f5

Reply:
Ownership is on VO side, but why not make sure we don't leak some stray pointer after reconfig/deinit? I though https://github.com/mpv-player/mpv/commit/2893b7d0f51094bfee0cc27f3bebba8de0fd8018 was about memory ownership.

vo->target_params is a weak pointer that has to be initialized on each reconfig by the vo.

Reply:
removed now

Reply:
It was primarily about the memory ownership yes but I also don't see a reason to force a specific lifetime onto the VO impl.